### ISSUE TITLE
GradNorm Spike in continuous-time scm training

### ISSUE THREAD

**Issue by zzkk-spec** [NVlabs/rcm#10]
Great work! When training continuous-time SCM (with fd_type=0, wo dmd) on the SD3.5 MMDiT architecture, training collapses — there is a spike in gradients, followed by the model outputting pure noise. Could you please advise on possible solutions to this issue?

<img width="365" height="361" alt="Image" src="https://github.com/user-attachments/assets/845c7188-3b47-4303-ada0-3de974e31df9" />

<img width="523" height="321" alt="Image" src="https://github.com/user-attachments/assets/99d967f0-9bab-4b44-a291-a1127707e74f" />

The curve below is the discrete-time CM (with fd_type=2, without dmd), and its gradient norm is relatively stable.

**Comment by zhengkw18** [comment]
Did you try adopting the high-precision time strategy, i.e., change all time embedding computation to fp32?

We encountered similar instability on Cosmos T2I models and enhanced stability by high-precision or semi-continuous time.

**Comment by zzkk-spec** [comment]
> Did you try adopting the high-precision time strategy, i.e., change all time embedding computation to fp32?
> 
> We encountered similar instability on Cosmos T2I models and enhanced stability by high-precision or semi-continuous time.

I resolved the collapse issue of sCM using semi-discrete and fp32 timestep embedding. Thank you for your suggestions!

**Comment by zhengkw18** [comment]
Great to hear that. You can also write pytest scripts like `rcm/networks/wan2pt1_jvp_test.py` to test the correctness and accuracy of JVP computation.
